File: leetcode/657_Judge_Route_Circle.py
# easy 
class Solution(object):
    def judgeCircle(self, moves):
        """
        :type moves: str
        :rtype: bool
        """
        if len(moves) == 0:
            return True
        i = 0
        j = 0
        for move in moves:
            if move == "R": i += 1
            if move == "L": i -= 1
            if move == "U": j += 1
            if move == "D": j -= 1
        if (i == 0 and j == 0): return True
        return False
        
# collections counter        
def judgeCircle(self, moves):
    c = collections.Counter(moves)
    return c['L'] == c['R'] and c['U'] == c['D']

File: leetcode/657_Judge_Route_Circle.java
    public boolean judgeCircle(String moves) {
        int x = 0, y = 0;
        for(char c: moves.toCharArray()){
            x += (c=='R'?1:0) + (c=='L'?-1:0); y += (c=='U'?1:0) + (c=='D'?-1:0);
        }
        return x == 0 && y == 0;
    }
    
class Solution {
    public boolean judgeCircle(String moves) {
        if (moves == null || moves.length() == 0)
            return true;
        int[] count = new int[2];
        for (char c : moves.toCharArray()) {
            if (c == 'U') count[0]++;
            if (c == 'D') count[0]--;
            if (c == 'L') count[1]++;
            if (c == 'R') count[1]--;
        }
        for (int i = 0; i < count.length; i++) 
            if (count[i] != 0) return false;
        return true;
    }
}